lib/Doctrine/DBAL/Portability/Statement.php

```diff
@@ -213,6 +213,55 @@ protected function fixRow($row, $iterateRow, $fixCase)
     }
 
     /**
+     * Fixes column values according to portability settings
+     *
+     * @param mixed[] $values Column values indexed by row number
+     * @param bool $emptyToNull Should empty strings be converted to NULL
+     * @param bool $rTrim Should strings be right-trimmed
+     * @return mixed[]
+     */
+    private function fixColumn(array $values, $emptyToNull, $rTrim)
+    {
+        if (!$emptyToNull && !$rTrim) {
```

Comment:
Isn't it simpler to loop twice, and call two separate methods that do one the `emptyToNull` fixing, and the other the `rTrim` fixing?